src/mantine-core/src/components/Tooltip/FloatingTooltip/FloatingTooltip.tsx

```diff
@@ -115,25 +120,41 @@ export const FloatingTooltip = forwardRef<HTMLDivElement, FloatingTooltipProps>(
     const [opened, setOpened] = useState(false);
     const { ref: mouseRef, x, y } = useMouse();
     const visible = opened && !disabled;
-    const [referenceElement, setReferenceElement] = useState(null);
-    const mergedRefs = useMergedRef(ref, setReferenceElement, mouseRef);
+    const mergedRefs = useMergedRef(ref, mouseRef);
     const coordinates = useMemo(() => {
       // There's no way to get the exact size of the
       // cursor using JS, however most OS's use 32x32
       const estimatedCursorSize = 32;
-      const tooltipWidth = _tooltipRef.current?.offsetWidth || 0;
+      const tooltipWidth =
+        typeof width === 'number' ? width : _tooltipRef.current?.offsetWidth || 0;
+      const tooltipHeight = _tooltipRef.current?.offsetHeight || 0;
 
       switch (position) {
         case 'top':
-          return { x: x - tooltipWidth / 2, y: y - estimatedCursorSize };
+          return {
+            left: x - tooltipWidth / 2,
+            top: y - tooltipHeight,
+          };
         case 'left':
-          return { x: x - estimatedCursorSize / 2 - tooltipWidth, y };
+          return {
+            left: x - tooltipWidth - estimatedCursorSize / 2,
+            top: y,
+          };
         case 'right':
-          return { x: x + estimatedCursorSize / 2, y };
+          return {
+            left: x + estimatedCursorSize / 2,
+            top: y,
+          };
         case 'bottom':
-          return { x: x - tooltipWidth / 2, y: y + estimatedCursorSize };
+          return {
+            left: x - tooltipWidth / 2,
+            top: y + tooltipHeight,
+          };
         default:
-          return { x: x || 0, y: y || 0 };
+          return {
+            left: x || 0,
+            top: y || 0,
+          };
       }
     }, [x, y]);
```

Comment:
Would adding the `positionDependencies` prop to the dependency array here effectively be the same?

Reply:
I do not really know, never dug deep into FloatingTooltip component as it was contributed. If it works, then it's fine.

Reply:
Yeah, I contributed it after all! So I guess I know best then 😄?! 

I added it back, and spread it into the dependency array, so the position will be recalculated if one of those dependencies changes.

Reply:
Well, let's hope that it will fix it then